src/main/java/net/dv8tion/jda/api/interactions/callbacks/IMessageEditCallback.java

```diff
@@ -0,0 +1,249 @@
+ *
+ * <p>Editing a message using these methods will automatically acknowledge the interaction.
+ *
+ * <h2>Deferred Edits</h2>
+ *
+ * Similar to {@link IReplyCallback}, message edits can be deferred and performed later with {@link #deferEdit()}.
+ * A deferred edit tells Discord, that you intend to edit the message this interaction was performed on, but will do so later.
+ * However, you can defer the edit and never do it, which is effectively a no-operation acknowledgement of the interaction.
+ *
+ * <p>If an edit is {@link #deferEdit() deferred}, it becomes the <b>original</b> message of the interaction hook.
+ * This means all the methods with {@code original} in the name, such as {@link InteractionHook#editOriginal(String)},
+ * will affect that original message you edited.
+ */
+public interface IMessageEditCallback extends IDeferrableCallback
+{
+    /**
+     * No-op acknowledgement of this interaction.
+     * <br>This tells discord you intend to update the message that the triggering component is a part of using the {@link #getHook() InteractionHook} instead of sending a reply message.
+     * You are not required to actually update the message, this will simply acknowledge that you accepted the interaction.
+     *
+     * <p><b>You only have 3 seconds to acknowledge an interaction!</b>
+     * <br>When the acknowledgement is sent after the interaction expired, you will receive {@link net.dv8tion.jda.api.requests.ErrorResponse#UNKNOWN_INTERACTION ErrorResponse.UNKNOWN_INTERACTION}.
+     * <p>Use {@link #editMessage(String)} to edit it directly.
+     *
+     * @return {@link MessageEditCallbackAction} that can be used to update the message
+     *
+     * @see    #editMessage(String)
+     */
+    @Nonnull
+    @CheckReturnValue
+    MessageEditCallbackAction deferEdit();
+
+    /**
+     * Acknowledgement of this interaction with a message update.
+     * <br>You can use {@link #getHook()} to edit the message further.
+     *
+     * <p><b>You can only use deferEdit() or editMessage() once per interaction!</b> Use {@link #getHook()} for any additional updates.
+     *
+     * <p><b>You only have 3 seconds to acknowledge an interaction!</b>
+     * <br>When the acknowledgement is sent after the interaction expired, you will receive {@link net.dv8tion.jda.api.requests.ErrorResponse#UNKNOWN_INTERACTION ErrorResponse.UNKNOWN_INTERACTION}.
+     *
+     * @param  message
+     *         The new message content to use
+     *
+     * @throws IllegalArgumentException
+     *         If the provided message is null
+     *
+     * @return {@link MessageEditCallbackAction} that can be used to further update the message
+     */
+    @Nonnull
+    @CheckReturnValue
+    default MessageEditCallbackAction editMessage(@Nonnull Message message)
+    {
+        Checks.notNull(message, "Message");
+        MessageEditCallbackActionImpl action = (MessageEditCallbackActionImpl) deferEdit();
+        return action.applyMessage(message);
```

Comment:
Because `applyMessage` is currently only available in the internal impl class. This is by design so we can more easily add support for edit message builders later.